### ISSUE TITLE
mat: ElementWiseAdd

### ISSUE THREAD

**Issue by glycerine** [gonum/gonum#881]
go1.10.7 on OSX, using gonum/v1/gonum/mat version 01c8581f3ecbaee4d9d7798b3a3a40daaa1bdd36 , the latest as of 2019 March 1.

I'm new to gonum, although I'm not new to numerical coding. I'm translating the matlab line "x = x + 1", where x is a matrix. In matlab, this does element-wise addition of 1 to all elements in the matrix x. 

I am surprised that I don't see an element-wise addition operator in the gonum Dense matrix API. Is it hidden somewhere?  Or just a minor oversight?  Or a deliberate (however odd) omissions? This seems like a basic operation that should be available without having to write (the probably slower than it needs to be)

~~~
func ElementWiseAdd(x *mat.Dense, addme float64) {
    x.Apply(func(i, j int, v float64) float64 {
        return v + addme
    }, x)
}
~~~

**Comment by kortschak** [comment]
I would probably just write that as

```
func addElem(x *mat.Dense, f float64) {
    r, c := x.Dims()
    for i := 0, i < r; i++ {
        for j := 0; j < c; j++ {
            x.Set(i, j, x.At(i, j) + f)
        }
    }
}
```

The alternative is to provide a Ones-like type that you can just `x.Add(x, ones)`. The ones-like type can be very space efficient, requiring only rows, cols and a value.

I'd like to hear how others address this.

**Comment by chewxy** [comment]
For `*Dense` I use `RawMatrix().Data`, then loop through it. I have found that to be more more performant than `.At(i,j)` and `Set`. 

Gorgonia's tensors do something similar in the background when doing these sorts of "broadcasted" operations

**Comment by kortschak** [comment]
Yes, we do bounds checks on `At` and `Set` so calling those will be more expensive than it would be if we did it internally where we'd use `at` and `set` which don't do bounds checks except when we turn on internal bounds checks for testing purposes.

**Comment by Kunde21** [comment]
FYI:  We have an asm-backed helper: `floats.AddConst`
